feat(ingestion): ingestion exclusion rule as a resource

jorgebay opened the pull request:
Expose a `logdna_ingestion_exclusion` resource to manage ingestion exclusions for an account.

Ref: LOG-10494
Semver: minor

jorgebay commented:
Integration tests against prod are failing because ingestion exclusion rules management is not yet in prod, it should be good to be reviewed though.

david-ly commented on logdna/exclusion_rule.go:
```diff
@@ -0,0 +1,49 @@
+		Optional:     true,
+		AtLeastOneOf: exclusionRuleAtLeastOneOfFields,
+	},
+}
```
Nit: needs a newline here

david-ly reviewed (commented):
lgtm, nice job on the schema + testing method modularizations! I thought we opted to make the `title` property optional for exclusion rules, did these updates intentionally change it to be required?

jorgebay commented on logdna/exclusion_rule.go:
```diff
@@ -0,0 +1,49 @@
+	},
+	"title": {
+		Type:     schema.TypeString,
+		Required: true,
```
Crap, I missed that one.

Nice catch @david-ly

mdeltito commented on docs/resources/logdna_ingestion_exclusion.md:
```diff
@@ -0,0 +1,39 @@
+
+The following arguments are supported by `logdna_ingestion_exclusion`:
+
+_Note:_ A `title` and at least one of the following properties: `apps`, `hosts`, `query` must be specified to create this resource.
+
+- `title`: **string** _(Required)_ Title of this exclusion rule that will appear in the UI.
```
already called out by @david-ly in the code, just noting this is still marked required in docs as well

mdeltito commented on logdna/provider_test.go:
```diff
@@ -8,6 +8,7 @@ import (
 )
 
 var serviceKey = os.Getenv("SERVICE_KEY")
+var apiHostUrl = os.Getenv("API_URL")
```
+1 this is helpful and I had been meaning to do this. we could also consider adding a secondary set of credentials in CI and running non-`main` branches against staging

jorgebay commented:
Thanks @david-ly & @mdeltito for the review, I'll update them soon.

Mike! I've based mostly on your implementation, it was simple to follow so kudos to you :)

jorgebay commented:
I've rebased and updated it to have the `title` field as optional.

Tests are failing on Jenkins but passing with the dev version of the API. In the meantime, it's ready to be reviewed.
